libarchive/archive_string.c

```diff
@@ -552,6 +552,8 @@ archive_wstring_append_from_mbs_in_codepage(struct archive_wstring *dest,
 		} else
 			mbflag = MB_PRECOMPOSED;
 
+		mbflag |= MB_ERR_INVALID_CHARS;
```

Comment:
This is to fix `test_entry` which expects that invalid UTF-8 characters produce errors, so I'm assuming that this is the expected behavior. Note that there are other locations in this file that fail with invalid characters.

https://github.com/libarchive/libarchive/blob/9ba8292aafacb7de8ff3629e823a862aa2ec569f/libarchive/test/test_entry.c#L921-L924